src/Sema.zig

```diff
@@ -8571,6 +8590,14 @@ fn coerce(
             // T to E!T or E to E!T
             return sema.wrapErrorUnion(block, dest_type, inst, inst_src);
         },
+        .ErrorSet => {
+            // error.Joe to anyerror
+            if (dest_type.tag() == .anyerror and inst_ty.tag() == .error_set_single) {
```

Comment:
`tag()` is the wrong abstraction level for Sema to be using here. Rather than checking how the type is *represented* which is what `tag()` provides, this code needs to ask the Type whether it is the global error set or not.

For checking `inst_ty`, use `zigTypeTag()` to make sure it is an error set, rather than only checking if it is represented as `error_set_single` - because error set values could be represented with other tag()s.